### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by sanieni6**
# Ruby on Rails capstone project - Budget app 💯 
The Ruby on Rails capstone project is about building a mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
## Demo
[Render Link](https://budgetapp-iykk.onrender.com/)
[Video Demo Link](https://www.loom.com/share/c845dcdc547a434cba389d742520625a?sid=739afb51-1f3a-44dc-8696-18090e29ecb3)
#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.
#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/shared/_header.html.erb [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+<% elsif name== 'Transactions' || name == 'New Category' %>
+    <%= link_to '', groups_path, class: 'back-image ml-1', style: "background-image: url(#{asset_path('arrow-small-left.svg')})" %>
+<% elsif name== 'Transactions' || name == 'New Transaction' %>
+    <%= link_to '', group_path(@group), class: 'back-image ml-1', style: "background-image: url(#{asset_path('arrow-small-left.svg')})" %>
 <% else %>
-    <%= link_to '', root_path, class: 'back-image ml-1', style: "background-image: url(#{asset_path('arrow-small-left.svg')})" %>
+    <%= link_to '',root_path, class: 'back-image ml-1', style: "background-image: url(#{asset_path('bar.svg')})" %>
 <% end %>
 <h1 class="font-nova text-lg flex flex-row items-center">
 <img src=<%=asset_path('wallet.svg')%> class="w-5 h-5"><%=name%></h1>
```
Comment (andersonlebon) [inline app/views/shared/_header.html.erb]: - [ ] By following the design requirements of this project, the welcome page does not have a hearer. kindly remove it.
image

- [ ] Kindly fix this header in your order by following the design guidlines, use the same colors, font-weight, margins, and padding

your header | the design's header
--|--
Screenshot 2023-07-06 at 07 35 36 | Screenshot 2023-07-06 at 07 33 03

- [ ] Kindly make sure the user can logout, you can replace the search icon in the header by a logout button and implement the logic in your app.

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -21,7 +19,4 @@
         <% end %>
         <button class="w-11/12 bg-main text-center p-2 text-white rounded font-nova mt-3 mb-3"> <%= link_to 'New Category', new_group_path %> </button>
         </ul>
```
Comment (andersonlebon) [inline app/views/groups/index.html.erb]: - [ ] Kindly make sure that the this button is style like the template one(color, padding, margin, width, background-color, position)

your design | the template
---|---
image | image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @sanieni6 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

almost-there

## Highlights 🍾 
- Nice use o Tailwinds Css 👏  ✔️ 
- Your Repo is very clean, clean readme, clean PR description ✔️ 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/shared/_header.html.erb (hunk 1 above)
Comment (sanieni6) [reply app/views/shared/_header.html.erb]: image

image

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (sanieni6) [reply app/views/groups/index.html.erb]: image

**Comment by sanieni6** [comment]
# 🟢 Changes Applied

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**